pyproject.toml

```diff
@@ -35,7 +35,7 @@ dependencies = [
   "setuptools==75.6.0",
   'snowflake.core==1.0.2; python_version < "3.12"',
   "snowflake-connector-python[secure-local-storage]==3.12.3",
-  'snowflake-snowpark-python>=1.15.0;python_version < "3.12"',
+  'snowflake-snowpark-python>=1.15.0,<1.26.0;python_version < "3.12"',
```

Comment:
not entirely sure why the context.py alone wasn't sufficient, but I had to do this too for E2E to pass.